### ISSUE TITLE
Value Checker should warn when @StaticallyExecutable is ignored

### ISSUE THREAD

**Issue by mernst** [typetools/checker-framework#1227]
A `@StaticallyExecutable` annotation is ignored unless the method's types are among those analyzed by the Value Checker.  If a user writes a `@StaticallyExecutable` annotation that will be ignored, then the Value Checker should issue a warning (not an error).

Here is example code for which the Index Checker issues a warning, because the Value Checker is not able to statically analyze the code, but that reason is not obvious to a programmer:

```java
import org.checkerframework.common.value.qual.ArrayLen;
import org.checkerframework.common.value.qual.StaticallyExecutable;
import org.checkerframework.dataflow.qual.SideEffectFree;

public class StaticallyExecutableSplitLines {

  public String /*@ArrayLen(11)*/[] testSplitLines() {
    return splitLines("one\ntwo\n\rthree\r\nfour\rfive\n\n\nsix\r\n\r\n\r\n");
  }

  /*@SideEffectFree*/ /*@StaticallyExecutable*/
  public static String[] splitLines(String s) {
    return s.split("\r\n?|\n\r?", -1);
  }

}
```
